### PR TITLE
Add JS utility unit tests with Jest

### PR DESCRIPTION
**Pull request by SeinopSys**
### Before you begin

* I understand my contributions may be rejected for any reason
* I understand my contributions are for the benefit of Derpibooru and/or the Philomena software
* I understand my contributions are licensed under the GNU AGPLv3

- [x] I understand all of the above

---

The goal of this PR to achieve unit tests coverage in the utility JavaScript files of the application in preparation for migration to TypeScript. This is to ensure no existing functionality will be broken by this process.

Running tests has also been added to CI, however, coverage requirements are only enforced inside the `assets/js/utils` folder. Once coverage reaches a certain threshold it may be worth raising the global minimum percentage, but for now it should be kept at 0 to allow gradual introduction of unit tests.

### CODE REVIEW

File: assets/js/utils/__tests__/tag.spec.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+      spoiler_image_uri: null,
+    },
+    5: {
+      id: 5,
+      name: 'lilo & stitch',
+      images: 6,
+      spoiler_image_uri: null,
+    },
+  };
+  const getEnabledSpoilerType = () => getRandomArrayItem<SpoilerType>(['click', 'hover']);
```
Comment (liamwhite) [inline assets/js/utils/__tests__/tag.spec.ts:48]: `static`?

File: assets/js/utils/store.js [hunk 2]
```diff
@@ -56,11 +58,8 @@ export default {
     const lastUpdatedKey = key + lastUpdatedSuffix;
     const lastUpdatedTime = this.get(lastUpdatedKey);
 
-    if (Date.now() > lastUpdatedTime) {
-      return true;
-    }
-
-    return false;
+    const currentTime = Date.now();
+    return currentTime > lastUpdatedTime;
```
Comment (liamwhite) [inline assets/js/utils/store.js:62]: `return Date.now() > lastUpdatedTime` ?

File: assets/js/utils/__tests__/image.spec.ts [hunk 3]
```diff
@@ -0,0 +1,585 @@
...
+      testSpoilerThumb(['click', 'mouseLeave']);
+    });
+
+    it('should add mouseenter and mouseleave handlers for hover spoiler type', () => {
+      window.booru.spoilerType = 'hover';
+      expect.hasAssertions();
+      testSpoilerThumb(['mouseEnter', 'mouseLeave']);
+    });
+
+    it('should not add event handlers for unknown spoiler type', () => {
```
[the remark this answers is not in the document]
Comment (SeinopSys) [reply assets/js/utils/__tests__/image.spec.ts:529]: Nope, I forgot about that spoiler type, I will add it to the types and tests.

File: assets/js/utils/__tests__/local-autocompleter.spec.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
+import { LocalAutocompleter } from '../local-autocompleter';
+import { promises } from 'fs';
+import { join } from 'path';
+import { TextDecoder, TextEncoder } from 'util';
```
[the remark this answers is not in the document]
Comment (SeinopSys) [reply assets/js/utils/__tests__/local-autocompleter.spec.ts:4]: Most likely just a leftover import, I can get rid of it

**Review by liamwhite — APPROVED** [review]

**Review by Meow — APPROVED** [review]
Looks good to me from just looking at it

**Pull request merged**